src/Sylius/Behat/Page/Shop/Account/AddressBook/IndexPage.php

```diff
@@ -41,97 +35,72 @@ public function getAddressesCount()
+        return $this->hasElement('messages');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function addressOfContains($fullName, $value)
+    public function addressOfContains(string $fullName, string $value): bool
     {
-        $address = $this->getAddressOf($fullName);
+        $address = $this->getAddressPageOf($fullName);
 
-        return $address->has('css', sprintf('address:contains("%s")', $value));
+        return $address->has('css', sprintf('[data-test-address]:contains("%s")', $value));
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function editAddress($fullName)
+    public function editAddress(string $fullName): void
     {
-        $addressToEdit = $this->getAddressOf($fullName);
-        $addressToEdit->findLink('Edit')->press();
+        $addressToEdit = $this->getAddressPageOf($fullName);
+        $addressToEdit->find('css', '[data-test-edit-button] [data-test-button]')->press();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function deleteAddress($fullName)
+    public function deleteAddress(string $fullName): void
     {
-        $addressToDelete = $this->getAddressOf($fullName);
-        $addressToDelete->pressButton('Delete');
+        $addressToDelete = $this->getAddressPageOf($fullName);
+        $addressToDelete->find('css', '[data-test-delete-button]')->press();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function setAsDefault($fullName)
+    public function setAsDefault(string $fullName): void
     {
-        $addressToSetAsDefault = $this->getAddressOf($fullName);
-        $addressToSetAsDefault->pressButton('Set as default');
+        $addressToSetAsDefault = $this->getAddressPageOf($fullName);
+        $addressToSetAsDefault->find('css', '[data-test-set-as-default-button]')->press();
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function hasNoDefaultAddress()
+    public function hasNoDefaultAddress(): bool
     {
         return !$this->hasElement('default_address');
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function getFullNameOfDefaultAddress()
+    public function getFullNameOfDefaultAddress(): string
     {
-        $fullNameElement = $this->getElement('default_address')->find('css', 'address > strong');
+        $fullNameElement = $this->getElement('default_address')->find('css', '[data-test-name]');
 
         Assert::notNull($fullNameElement, 'There should be a default address\'s full name.');
 
         return $fullNameElement->getText();
     }
 
-    /**
-     * @param string $fullName
-     *
-     * @return NodeElement|null
-     */
-    private function getAddressOf($fullName)
+    private function getAddressOf(string $fullName): ?NodeElement
     {
-        return $this->getElement('addresses')->find('css', sprintf('div.address:contains("%s")', $fullName));
+        return $this->getElement('addresses')->find('css', sprintf('[data-test-address]:contains("%s")', $fullName));
```

Comment:
```suggestion
        return $this->getElement('addresses')->find('css', sprintf('[data-test-address="%s"]', $fullName));
```
and add in html : `sylius_test_html_attribute('address', <variable with full name :D>)`

Reply:
ofc here and everywhere where we have a similar case